### ISSUE TITLE
[QUESTION] Сhoosing a metric to save the best checkpoint

### ISSUE THREAD

**Issue by DKAvocad00** [lightly-ai/lightly-train#578]
### 🤔 What’s your question?

Hi. Can i some how change the base metric to save the best checkpoint during lightly_train.train_semantic_segmentation() ? 

when i trained my mode i found that lightly_train semantic segmentation saving best checkpoint according to mIoU and best checkpoint like stopping before overfiting. But with base metrics mIoU it doesn't work well.

I have the case when my model overfit during the train process val loss increased in  the same time mIoU increased too, which not good at all.  

Thats why, can i save my model based on loss, recall, precision or something else based on task ?

**Comment by stegmuel** [comment]
Thanks.
The segmentation method is [EoMT](https://arxiv.org/pdf/2503.19108). This method has the particularity that it uses masking during training to restrict the tokens a given query is allowed to attend to. This technique allows faster convergence, but is slower at inference as flash-attention can't naively be used with masks. To that end, EoMT introduces mask annealing during training to progressively get rid of the dependence on masks. 
As opposed to the official EoMT implementation we decided to perform the validation step without masks to have access to the real performance of the model. Due to this annealing process and the mismatch between training and validation behavior it is normal to see the validation loss increase and it is not a sign of overfitting. 
I would recommend sticking to mIoU and in general to let the training terminate to ensure that the mask annealing process is done.

**Comment by guarin** [comment]
@DKAvocad00 are you fine-tuning from one of the LightlyTrain semantic segmentation models trained on COCO/Cityscapes?

If yes you might want to disable the mask annealing that @stegmuel mentioned with:
```
lightly_train.train_semantic_segmentation(
    ...,
    model_args={
        "attn_mask_annealing_steps_start": [0, 0, 0],
        "attn_mask_annealing_steps_end": [0, 0, 0],
    }
)
```

And to answer your original question, you can configure the metrics for saving the best checkpoint with:
```
lightly_train.train_semantic_segmentation(
    ...,
    save_checkpoint_args={
        "watch_metric": "val_loss",   # or any other metric name like "val_metric/miou"
    }
)
```

**Comment by DKAvocad00** [comment]
@stegmuel @guarin Thanks a lot. I trained raw dinov3-eomt 
```
lightly_train.train_semantic_segmentation(
        out="...",
        model="dinov3/vits16-eomt",
        ...
)
```

**Comment by guarin** [comment]
Alright, then you need the masked annealing :)
